src/local-history/local-history-manager.ts

```diff
@@ -74,22 +89,22 @@ export class LocalHistoryProvider {
             }
 
             // Limits the number of entries based on the preference 'local-history.maxEntriesPerFile`.
-            items = items.slice(0, this.maxEntriesPerFile);
+            items = items.slice(0, this.localHistoryPreferenceService.maxEntriesPerFile);
             vscode.window.showQuickPick(items,
                 {
                     placeHolder: `Please select a local history revision for '${path.basename(textEditor.document.fileName)}'`,
                     matchOnDescription: true,
                     matchOnDetail: true
                 }).then((selection) => {
-                    // User made final selection
+                    // User made final selection.
                     if (!selection) {
                         return;
                     }
 
-                    // Get the file system path for the selection
+                    // Get the file system path for the selection.
                     const selectionFsPath = path.join(`${workspaceFolderPath}`, '.local-history', selection.label);
```

Comment:
Since `selection.label` is now the calendar icon and date+time and we now store the file path in `selection.description`, we should update this line to
```ts 
const selectionFsPath = path.join(`${workspaceFolderPath}`, '.local-history', path.basename(selection.description!));
```

Reply:
> Since `selection.label` is now the calendar icon and date+time and we now store the file path in `selection.description`, we should update this line to
> 
> ```ts
> const selectionFsPath = path.join(`${workspaceFolderPath}`, '.local-history', path.basename(selection.description!));
> ```

Please open an issue so you can fix it :)